### ISSUE TITLE
Adding option `echo = FALSE` for text to R Markdown

### ISSUE THREAD

**Issue by lauracion** [ropensci/unconf18#74]
## Summary for another of the projects that came up while discussing #42 

@jtr13 wrote: "there's one small piece of the puzzle that I doubt would be hard to implement and would make a big difference. That is, having an `echo=FALSE` option for text, to provide the same flexibility for text in progress as we have with code in progress. I can think of so many uses: the ability for example to create assignments with and without solutions (... there are workarounds using comments in code chunks but that's not the same)."

**Comment by zachary-foster** [comment]
Adding knitr engine might work:

https://yihui.name/knitr/demo/engines/

**Comment by zachary-foster** [comment]
Oh wait, it already exists, you can use the `asis` chunk type to put markdown in chunks and use `echo = FALSE` to not include them. I just tried it and it works.

**Comment by lauracion** [comment]
That was a fast resolution! Thank you!

**Comment by lauracion** [comment]
That was a fast resolution! Thank you!

**Comment by zachary-foster** [comment]
No problem!

**Comment by jtr13** [comment]
Thanks... The problem though with `asis` is that you still need `cat()` which is a pain. In addition, with math equations you have to double escape the tex stuff, which is difficult. (See here:  https://community.rstudio.com/t/echo-false-type-option-for-rmarkdown-text/2384 -- this has been on my mind for a while!)

I have in mind being able to write markdown paragraphs that are included or not, without having to wrap each line in r code.

**Comment by zachary-foster** [comment]
I dont think you need `cat`. I meant the chunk engine `asis`, not  the chunk option `results = "asis"`. Like so:

```

# Show

```{asis  echo = FALSE}
## Something entirely not thought out

I really would rather people not see this yet.
``'

# Dont show

```{asis  echo = FALSE}
## Something entirely not thought out

I really would rather people not see this yet.
``'
```

**Comment by jtr13** [comment]
Aha... brilliant! Where can I apply to get back the time I lost on workarounds??? :-) Also, is that documented anywhere? Not here: https://yihui.name/knitr/demo/engines/

**Comment by zachary-foster** [comment]
>Where can I apply to get back the time I lost on workarounds???

Haha, there is probably a form you can fill out somewhere. 

> Also, is that documented anywhere?
 
Not that I saw. I had to look through the source code to find it: 

https://github.com/yihui/knitr/blob/dc028f4c9698f84999b53edc5f6f255b29d7e5a2/R/engine.R#L390-L392

**Comment by jtr13** [comment]
Ok, so I hereby change this issue to providing documentation for the `asis` engine!

**Comment by apreshill** [comment]
A related tool I was just made aware of from issue #63 is `assignr`: https://github.com/coatless/assignr

I have not used, but looks promising @jtr13!

**Comment by lauracion** [comment]
Summary: engine `asis` in `knitr` can be used to easily exclude from the final report chunks of text that are still in progress. `asis` needs to be documented.

**Comment by maurolepore** [comment]
Thank you @zachary-foster ! It'll be cool to tweet this on #rstats.

![image](https://user-images.githubusercontent.com/5856545/40273950-a708eb36-5b7f-11e8-88f9-c66eb53e5644.png)

**Comment by yihui** [comment]
It was only documented in two places: 1) the source code in `knitr:::eng_asis` (in the spirit of "Luke, use the source"); and 2) the knitr book, which is unfortunately not freely available like my other books, but you can still see the relevant text from Google Books: https://books.google.com/books?id=fyIuDwAAQBAJ&lpg=PP1&dq=dynamic%20documents%20with%20r%20and%20knitr&pg=PT145#v=onepage&q=asis%20engine&f=false

There are still many other features of knitr that are not publicly documented, and I plan to address this problem later this year but writing another book that will be free to read.

**Comment by lauracion** [comment]
:heart_eyes: Thank you, @yihui!!
